Assets/Tests/InventoryTests.cs

```diff
@@ -0,0 +1,154 @@
+                {InventoryItems.Fertilizer,0},
+                {InventoryItems.Stone,0}
+            };
+            
+            //Then
+            Assert.AreEqual(expected,inventory.GetInventory());
+        }
+    
+        [Test]
+        public void Add_OneResources_ReturnsSameNumber()
+        {
+            //Given
+            Inventory inventory = new Inventory();
+            
+            //When
+            inventory.AddToInventory(InventoryItems.Wood,1);
+            
+            //Then
+            Assert.AreEqual(1,inventory.GetInventoryAmount(InventoryItems.Wood));
+        }
+    
+        [Test]
+        public void Add_ZeroResources_ReturnsSameNumber()
+        {
+            //Given
+            Inventory inventory = new Inventory();
+            
+            //When
+            inventory.AddToInventory(InventoryItems.Wood,0);
+            
+            //Then
+            Assert.AreEqual(0,inventory.GetInventoryAmount(InventoryItems.Wood));
+        }
+    
+        [Test]
+        public void Remove_FromEmptyInventory_ReturnsZero()
+        {
+            //Given
+            Inventory inventory = new Inventory();
+            
+            //When
+            inventory.RemoveFromInventory(InventoryItems.Wood,1);
+            
+            //Then
+            Assert.AreEqual(0,inventory.GetInventoryAmount(InventoryItems.Wood));
+        }
+    
+        [Test]
+        public void Remove_FromFilledInventory_ReturnsLowerNumber()
+        {
+            //Given
+            Inventory inventory = new Inventory();
+            
+            //When
+            inventory.AddToInventory(InventoryItems.Wood,2);
+            inventory.RemoveFromInventory(InventoryItems.Wood,2);
+            
+            //Then
+            Assert.AreEqual(0,inventory.GetInventoryAmount(InventoryItems.Wood));
+        }
+    
+        [Test]
+        public void CanBuild_EmptyInventory_ReturnsFalse()
+        {
+            //Given
+            Inventory inventory = new Inventory();
+            
+            //Then
+            Assert.IsFalse(inventory.CanBuild(InventoryItems.Wood));
+        }
+    
+        [Test]
+        public void CanBuild_EnoughInInventory_ReturnsTrue()
+        {
+            //Given
+            Inventory inventory = new Inventory();
+            
+            //When
+            inventory.AddToInventory(InventoryItems.Wood,10);
+            
+            //Then
+            Assert.IsTrue(inventory.CanBuild(InventoryItems.Wood));
+        }
+    
+        [Test]
+        public void Event_InventoryUpdate_Add_Invoked()
+        {
+            //Given
+            Inventory inventory = new Inventory();
+            bool eventInvoked = false;
+            inventory.InventoryUpdate += (sender, args) => eventInvoked = true;
+        
+            //When
+            inventory.AddToInventory(InventoryItems.Wood,1);
+        
+            //Then
+            Assert.IsTrue(eventInvoked);
```

Comment:
Always use variables in asserts not methods. Makes debugging easier when needed.

Reply:
eventInvoked is a variable